src/Core/Injector/Injector.php

```diff
@@ -856,6 +856,8 @@ protected function setObjectProperty($object, $name, $value)
      *
      * Will recursively call itself for each depth of dotting.
      *
+     * @template T of object
+     * @param class-string<T>|string
      */
     public function has(string $name): bool
```

Comment:
I know I said no formal review process - this is just for my own understanding rather than a proper review.

What benefit does adding generics to this method provide? It doesn't seem like it would affect developer experience, since the return type is always going to be boolean regardless of what is passed in.

It looks like a lot of the generic types you've added are similar to this one so definitely keen to understand what the net effect of them is.

It seems to me that generics are mostly useful where they affect the return type - but I may have misunderstood something?

Reply:
You're right about this, I got a little excited and starting adding them everywhere - better to just keep them to the public facing methods

Reply:
I think I may have done the same thing to be fair 😅